### PR TITLE
DD-449 - Use "release migrated" function from dans-deposit-to-dataverse

### PR DESCRIPTION
**Pull request by ChrisMuller80**
Fixes DD-449

# Description of changes
added support for import of publication date

# How to test
1. Use latest archaeology base box
2. check out https://github.com/DANS-KNAW/dans-dataverse-scala-lib/pull/10 and run mvn clean install
3. check out this pr and run mvn clean install (this should add dans-dataverse-scala-lib to the dependencies)
4. check out  https://github.com/IQSS/dataverse/pull/7504 and run mvn clean install
5. redeploy .war file in dd-dtap
6. deploy module dd-dans-deposit-to-dataverse in dd-dtap
7. import a dataset with an amd file
8. Check the publication date

# Related PRs 
(Add links)
* https://github.com/DANS-KNAW/dans-dataverse-scala-lib/pull/10

# Notify
@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala [hunk 1]
```diff
@@ -82,12 +82,32 @@ case class DepositIngestTask(deposit: Deposit,
...
       _ <- instance.dataset(persistentId).awaitUnlock()
-      _ <- if (publish) publishDataset(persistentId)
+      publicationDate <- getPublicationdate(optAmd)
+      _ <- if (publish) publishDataset(persistentId, publicationDate)
            else keepOnDraft()
     } yield ()
     // TODO: delete draft if something went wrong
   }
 
+  def getPublicationdate(optAmd: Option[Node]): Try[String] = Try {
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala:92]: See: https://github.com/DANS-KNAW/dd-dans-deposit-to-dataverse/blob/e92fbccd9737b8aa7c456feddad74dfcc731c60a/src/main/scala/nl.knaw.dans.easy.dd2d/mapping/Amd.scala#L22

File: src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala [hunk 2]
```diff
@@ -82,12 +82,32 @@ case class DepositIngestTask(deposit: Deposit,
...
+    optAmd
+      .flatMap(amd => findOldestChangeDate(amd))
+      .getOrElse(LocalDate.now().toString)
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala:95]: I think we should use the normal publish API for non-migration deposits.

Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala:95]: It was my suggestion. Deposit date in emd/ddm is not the date it became public, at least not at EASY. You are saying this is meant for future deposits going straight into dataverse? I was not aware of that context.

File: src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala (hunk 1 above)
Comment (ChrisMuller80) [reply src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala]: Done

**Pull request merged**